common/setup.c

```diff
@@ -241,11 +238,14 @@ void __noreturn __text_init kernel_start(uint32_t multiboot_magic, unsigned long
             boot_flags.nosmp = true;
     }
 
+    init_ioapic();
+
+    /* Initialize timers */
+    init_timers(bsp);
+
     if (!boot_flags.nosmp)
         init_smp();
 
-    init_ioapic();
-
     init_pci();
```

Comment:
Can you please explain (in the commit log) why that specific order is needed, e.g. why the IOAPIC needs to be initialized before the HPET or PIC? It feels counter-intuitive.